src/network/providers/provider.ts

```diff
@@ -0,0 +1,141 @@
+import EthereumProvider from './ethereum-provider';
+import SolanaProvider from './solana-provider';
+import {JsonRpcResponse} from 'hardhat/types';
+import {DomainRegistry} from '../../name_service/types';
+import logger from '../../core/log';
+import {RPCRequest} from '../../rpc/rpc-handlers';
+
+const log = logger.child({module: 'BlockchainProvider'});
+
+export enum ChainId {
+    PointYnet = 'ynet',
+    EthereumMainnet = 1,
+    EthereumRinkeby = 4,
+    SolanaMainnet = 'mainnet-beta',
+    SolanaDevnet = 'devnet'
+}
+
+export enum Blockchain {
+    ETHEREUM = 'ethereum',
+    SOLANA = 'solana'
+}
+
+/**
+ * All blockchain providers must comply with this interface.
+ */
+export interface Provider {
+    name(chainId: ChainId): string;
+    resolveDomain(chainId: ChainId, domain: string): Promise<DomainRegistry>;
+    setDomainContent(
+        chainId: ChainId,
+        domain: string,
+        data: string
+    ): Promise<Record<string, string|number|boolean>>;
+    send(chainId: ChainId, data: RPCRequest): Promise<JsonRpcResponse>;
+    sendTransaction(chainId: ChainId, reqId: number, txParams: unknown[]): Promise<JsonRpcResponse>;
+    requestAccounts(chainId: ChainId, reqId: number): Promise<JsonRpcResponse>;
+    getBlockNumber?(chainId: ChainId): Promise<number>;
```

Comment:
Methods that not all Providers implement are marked as optional. Then we check if it's implemented (comment below).